probe-rs/src/core.rs

```diff
@@ -218,9 +219,80 @@ impl<'probe> MemoryInterface for Core<'probe> {
     }
 }
 
+/// A struct containing key information about an exception.
+/// The exception details are architecture specific, and the abstraction is handled in the
+/// architecture specific implementations of [`crate::core::ExceptionInterface`].
+pub struct ExceptionInfo {
+    /// A human readable explanation for the exception.
+    pub description: String,
+    /// The stackframe registers, and their values, for the frame that triggered the exception.
+    pub calling_frame_registers: DebugRegisters,
+}
+
+/// A generic interface to identify and decode exceptions during unwind processing.
+pub trait ExceptionInterface {
+    /// Using the `stackframe_registers` for a "called frame",
+    /// determine if the given frame was called from an exception handler,
+    /// and resolve the relevant details about the exception, including the reason for the exception,
+    /// and the stackframe registers for the frame that triggered the exception.
+    /// A return value of `Ok(None)` indicates that the given frame was called from within the current thread,
+    /// and the unwind should continue normally.
+    fn get_exception_info(
+        &mut self,
+        _stackframe_registers: &DebugRegisters,
+    ) -> Result<Option<ExceptionInfo>, Error> {
+        // For architectures where the exception handling has not been implemented in probe-rs,
+        // this will result in maintaining the current `unwind` behavior, i.e. unwinding will stop
+        // when the first frame is reached that was called from an exception handler.
+        Ok(None)
+    }
+
+    /// Using the `stackframe_registers` for a "called frame", retrieve updated register values for the "calling frame".
+    fn calling_frame_registers(
+        &mut self,
+        _stackframe_registers: &crate::debug::DebugRegisters,
+    ) -> Result<crate::debug::DebugRegisters, crate::Error> {
+        Err(error::Error::Other(anyhow!(
```

Comment:
We have a dedicated variant for this.

Reply:
I can't find it. Can you point me to it please?

Reply:
https://github.com/probe-rs/probe-rs/blob/77413aa8c204b6d3472e419ab3943cfca16cf167/probe-rs/src/probe.rs#L142 it's on the DebugProbe only. It would be good if we had a variant for this on the respective error enum.

Reply:
Thanks ... I will add it.

Reply:
Done :)